### PR TITLE
HDDS-10061. NPE when container is loaded with missing container DB

### PR DESCRIPTION
**Pull request by sumitagrawl**
## What changes were proposed in this pull request?

when container db is not available, throw IOException so that its not loaded to memory. Log also represent for not loading the container but seems not handled properly.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10061

## How was this patch tested?

- Unit testcase added

### CODE REVIEW

**Review by errose28 — COMMENTED** [review]
Thanks for fixing this @sumitagrawl. Change LGTM but can you add a test as well to make sure that invalid containers missing a container file or DB are not loaded into the datanode on startup?

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/helpers/KeyValueContainerUtil.java [hunk 1]
```diff
@@ -232,7 +232,7 @@ public static void parseKVContainerData(KeyValueContainerData kvContainerData,
       LOG.error("Container DB file is missing for ContainerID {}. " +
           "Skipping loading of this container.", containerID);
       // Don't further process this container, as it is missing db file.
-      return;
+      throw new IOException("Container DB file is missing");
```
Comment (kerneltime) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/helpers/KeyValueContainerUtil.java:235]: Include the container ID in the exception.

**Review by errose28 — APPROVED** [review]
Thanks @adoroszlai LGTM once CI passes.

**Comment by adoroszlai** [comment]
Thanks @sumitagrawl for the patch, @errose28, @kerneltime for the review.

**Pull request merged**